packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -2116,10 +2154,31 @@ class CupertinoDialogAction extends StatelessWidget {
   /// Typically a [Text] widget.
   final Widget child;
 
-  /// Whether the button is enabled or disabled. Buttons are disabled by
-  /// default. To enable a button, set its [onPressed] property to a non-null
-  /// value.
-  bool get enabled => onPressed != null;
+  @override
+  State<CupertinoDialogAction> createState() => _CupertinoDialogActionState();
+}
+
+class _CupertinoDialogActionState extends State<CupertinoDialogAction>
```

Comment:
Is this going to be a leaf `_SlideTarget`? Since this is a public widget that takes a `Key`, I think it's possible for the `if (_currentTargets.firstOrNull != foundTargets.firstOrNull) ` check in `_updateDrag` to be false if a `CupertinoDialogAction` gets reparented while a "sliding" gesture is in progress?

Reply:
Yeah it's technically possible. In fact, if the action buttons somehow moves (for example, if a button at the top expands by itself so that the highlighted button is moved out of the pointer position), the dialog will not re-perform hit tests without a pointer event and will not update.

However, in reality I can't imagine when this would happen. Completely solving this require the same logic as `MouseTracker`, which additionally performs a hit test per frame. I agree that it'd be a good idea to re-use the code from `MouseTracker` (or, better put, to conditionally enable hovering for touch events). However that's a really rare edge case and I suggest we leave to a future PR, such as as part of https://github.com/flutter/flutter/issues/19786. What do you think?

Reply:
Maybe add a "TODO"? 

>  which additionally performs a hit test per frame

I thought we are already doing that in the current implementation (in `_updateDrag`)? We are just not comparing the hit-testing paths in a per-entry fashion.

Reply:
> Maybe add a "TODO"?

Sure!

> I thought we are already doing that in the current implementation

`_updateDrag` is only triggered every time the pointer moves. But if the app is updated without while the pointer stays still, then the pointer won't hit test the new layout. Really tricky situation.